macros/dimensional_dbt_internals/abstraction.sql

```diff
@@ -5,18 +5,32 @@
 
 {%- endmacro -%}
 
-{%- macro after_select(source_ctes, column_count) -%}
+{%- macro after_select(source_ctes, column_count, partial) -%}
+    {% if partial %}
+        ,{{ dimensional_dbt.coalesce_snapshot_cols(source_ctes, 'valid_from') }} AS dbt_valid_from
+        ,{{ dimensional_dbt.coalesce_snapshot_cols(source_ctes, 'valid_to') }} AS dbt_valid_to
+        ,{{ dimensional_dbt.coalesce_snapshot_cols(source_ctes, 'updated_at') }} AS dbt_updated_at
+        ,{{ dimensional_dbt.coalesce_snapshot_cols(source_ctes, 'scd_id') }} AS dbt_scd_id
+    {% else %}
         ,{{ dimensional_dbt.dim_columns() }}
+    {% endif %}
     FROM
-        {{ dimensional_dbt.from_clause(source_ctes, column_count) }}
+        {% set final_column_count = (column_count + 3) if partial else column_count %}
+        {{ dimensional_dbt.from_clause(source_ctes, final_column_count) }}
 )
 SELECT
     dimensional_dbt_column_selection.*
+    {% if not partial %}
     ,{{ dimensional_dbt.generate_dim_key() }}
+    {% endif %}
 FROM
     dimensional_dbt_column_selection
 ORDER BY
-    dim_valid_from, 
-    {{ dimensional_dbt.dim_key() }}
+    {% if partial %}
```

Comment:
Where is the docs for this syntax? (e.g., if partial)

Reply:
It is just a conditional statement where `partial` is a boolean variable, if I understand correctly

Reply:
ah, so partial is a column and not part of the language?

Reply:
yep, it's a param you can set when calling the macro. so basically, you can say `partial=true` and it will build the resulting table to look like a DBT snapshot (and not a dimensional_dbt dim, which is the final product). then you can lace these together with other DBT snapshots (or other partials) to build your final dimensional_dbt dim ;)